front_end/src/App.js

```diff
@@ -0,0 +1,106 @@
+import Products from './components/Products';
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { fetchProducts } from './actions/productActions';
+import Pagination from './components/Pagination';
+import 'bootstrap';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import 'bootstrap/dist/js/bootstrap.bundle.min.js';
+
+const App = () => {
+  const [price, setPrice] = useState('')
+  const [category, setCategory] = useState('')
+  const [query, setQuery] = useState('')
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchProducts())
+  }, [dispatch])
+
+  const handleCategoryClick = (e) => {
+    e.preventDefault()
+    console.log('Ctg')
+    setCategory(e.target.value)
+    dispatch(fetchProducts(price, e.target.value, query, currentPage));
+  }
+
+  const handlePrice = (e) => {
+    e.preventDefault()
+    console.log('Price')
+    setPrice(e.target.value)
+    dispatch(fetchProducts(e.target.value, category, query, currentPage));
+  }
+
+  const handleInputChange = (e) => {
+    e.preventDefault()
+    console.log('input')
+    setQuery(e.target.value);
+  }
+
+  const handleSearch = (e) => {
+    e.preventDefault()
+    setQuery(query);
+    dispatch(fetchProducts(price, category, query, currentPage));
+    console.log('query')
+  }
+
+  const handlePageChange = (page) => {
+    setCurrentPage(page);
+    dispatch(fetchProducts(price, category, query, page));
+    console.log('page')
+  }
+
+  return (
+    <div className="App justify-content-center">
+      <div className="navbar container-fluid justify-content-center">
+        <div className="row">
+          <div className="col-12">
+            <form onSubmit={handleSearch}>
```

Comment:
recommend keeping app.js as small as possible, the form area should be its own component